models/concerns/validate_data.rb

```diff
@@ -128,6 +128,12 @@ def check_services(schedule)
   def check_relation_consistent_timewindows
     unconsistent_relation_timewindows = []
     @hash[:relations].each{ |relation|
+      if Models::Relation::ON_VEHICLES_TYPES.include?(relation[:type]) && relation[:linked_ids].to_a.any?
+        raise OptimizerWrapper::DiscordantProblemError.new("#{relation[:type]} does not expect any service ids")
+      elsif Models::Relation::ON_SERVICES_TYPES.include?(relation[:type]) && relation[:linked_vehicle_ids].to_a.any?
+        raise OptimizerWrapper::DiscordantProblemError.new("#{relation[:type]} does not expect any vehicle ids")
```

Comment:
this check is not really about `relation_consistent_timewindows` maybe we can merge lapse_need checks and this vehicle/service_need checks into `check_relation_needs` or `check_relation_parameter_need`, `check_relation_argument_need` or something like that. Vehicle/service check should come first because we use linked_ids/linked_vehicle_ids in the lapse checks.